src/v2/params/index.ts

```diff
@@ -55,6 +54,16 @@ function registerParam(param: Param) {
   declaredParams.push(param);
 }
 
+/**
+ * For testing.
+ * @internal
+ */
+export function clearParams() {
+  while (declaredParams.length > 0) {
```

Comment:
I appreciate that you're trying to avoid letting constness escape the package, but fortunately JavaScript has another horrifying const-breaking method that lets you clear an array in O(1) instead of O(N):

```
declaredParams.splice(0, declaredParams.length);
```

Reply:
Huh. Done.